cmd/kubeadm/app/phases/copycerts/copycerts.go

```diff
@@ -191,15 +194,85 @@ func certsToUpload(cfg *kubeadmapi.InitConfiguration) map[string]string {
 	return certs
 }
 
-func getSecretData(cfg *kubeadmapi.InitConfiguration, key []byte) (map[string][]byte, error) {
+func getDataFromDisk(cfg *kubeadmapi.InitConfiguration, key []byte) (map[string][]byte, error) {
 	secretData := map[string][]byte{}
-	for certName, certPath := range certsToUpload(cfg) {
+	for certName, certPath := range certsToTransfer(cfg) {
 		cert, err := loadAndEncryptCert(certPath, key)
 		if err == nil || (err != nil && os.IsNotExist(err)) {
-			secretData[strings.Replace(certName, "/", "-", -1)] = cert
+			secretData[certOrKeyNameToSecretName(certName)] = cert
 		} else {
 			return nil, err
 		}
 	}
 	return secretData, nil
 }
+
+// DownloadCerts downloads the certificates needed to join a new control plane.
+func DownloadCerts(client clientset.Interface, cfg *kubeadmapi.InitConfiguration, key string) error {
```

Comment:
We need more unit tests here. Especially on this one.

Reply:
Yeah, is on the description of the PR. I was wondering if we can go with this one as it is, and I will create a follow up PR with unit tests for this, I just don't want to risk the deadline, and unit tests for an existing feature can span the 7th March deadline. WDYT?

Reply:
Yeah, I am fine with this. Ping me, when you rebase it, for lgtm.